jobs/cache_maintenance/src/cache_maintenance/metrics.py

```diff
@@ -0,0 +1,35 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 The HuggingFace Authors.
+
+import logging
+
+from libcommon.metrics import CustomMetric
+from libcommon.processing_graph import ProcessingStep
+from libcommon.queue import Queue
+from libcommon.simple_cache import get_responses_count_by_kind_status_and_error_code
+
+
+def collect_metrics(processing_steps: list[ProcessingStep]) -> None:
+    logging.info("collecting jobs metrics")
+    queue = Queue()
+    CustomMetric.objects(metric="queue_jobs_total").delete()
+    for processing_step in processing_steps:
+        for status, total in queue.get_jobs_count_by_status(job_type=processing_step.job_type).items():
+            CustomMetric(
+                metric="queue_jobs_total",
+                content={"queue": processing_step.job_type, "status": status, "count": total},
```

Comment:
maybe it's simpler to create a collection for every metrics, with an entry per queue for this metric, and an entry per cache kind in the other one

Reply:
(maybe it's not simpler, we would lose genericity)

Reply:
Makes sense, done with  separated collections